scripts/fetch-pr-artifacts.ts

```diff
@@ -0,0 +1,124 @@
+#!/usr/bin/env -S tea -E
+
+/*---
+args:
+  - deno
+  - run
+  - --allow-net
+  - --allow-env
+  - --allow-write=./artifacts.tgz
+  - --import-map={{ srcroot }}/import-map.json
+---*/
+
+/// Test
+/// ./scripts/fetch-pr-artifacts.ts e582b03fe6efedde80f9569403555f4513dbec91
+
+import { S3 } from "https://deno.land/x/s3@0.5.0/mod.ts";
+import { panic, undent } from "utils/index.ts";
+
+/// Main
+/// -------------------------------------------------------------------------------
+
+const usage = "usage: fetch-pr-artifacts.ts {REPO} {SHA} {platform+arch}"
+const repo = Deno.args[0] ?? panic(usage)
+const ref = Deno.args[1] ?? panic(usage)
+const flavor = Deno.args[2] ?? panic(usage)
+
+const res = await queryGraphQL<CommitQuery>(prQuery(repo))
+
+const node = res.repository?.ref?.target?.history?.edges.find(n => n.node.oid === ref)
+const pr = node?.node.associatedPullRequests.nodes[0].number
+
+const s3 = new S3({
+  accessKeyID: Deno.env.get("AWS_ACCESS_KEY_ID")!,
+  secretKey: Deno.env.get("AWS_SECRET_ACCESS_KEY")!,
+  region: "us-east-1",
+})
+const bucket = s3.getBucket(Deno.env.get("AWS_S3_BUCKET")!)
```

Comment:
I'd rather we didn’t use `dist.tea.xyz` for both staging and production.

Reply:
We don't. We pass in a different bucket in the build.yml.